src/Features/LanguageServer/ProtocolUnitTests/CodeActions/CodeActionsTests.cs

```diff
@@ -145,6 +136,48 @@ void M()
             Assert.Equal(AddImportDiagnosticIds.CS0103, addImport.Diagnostics.Single().Code.Value);
         }
 
+        [WpfTheory, CombinatorialData]
+        public async Task TestNestedCodeActions(bool mutatingLspWorkspace)
+        {
+            var markup = @"
+class ABC
+{
+    private static async void {|caret:XYZ|}()
+    {
+    }
+}";
+
+            await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace);
+
+            var caret = testLspServer.GetLocations("caret").Single();
+            var codeActionParams = new LSP.CodeActionParams
+            {
+                TextDocument = CreateTextDocumentIdentifier(caret.Uri),
+                Range = caret.Range,
+                Context = new LSP.CodeActionContext
+                {
+                    Diagnostics = new[]
+                    {
+                        new LSP.Diagnostic
+                        {
+                            // async method lack of await.
+                            Code = "CS1998"
+                        }
+                    }
+                }
+            };
+
+            var results = await RunGetCodeActionsAsync(testLspServer, codeActionParams);
+            Assert.Equal(7, results.Length);
+            Assert.Equal("Make method synchronous", results[0].Title);
```

Comment:
This would make the list hard to read unfortunately. Could we file an issue with LSP to add support for nested action?

Reply:
They have an issue for this, https://github.com/microsoft/language-server-protocol/issues/994
I will try push them to make progress on this